File: TestAndroid/app/src/main/java/com/example/testandroid/MainActivity.java
package com.example.testandroid;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private EditText edtMostar;
    private Button[] arrayButton = new Button[10];
    private Button btn1Numero;
    private Button btn2Numero;
    private Button btn3Numero;
    private Button btn4Numero;
    private Button btn5Numero;
    private Button btn6Numero;
    private Button btn7Numero;
    private Button btn8Numero;
    private Button btn9Numero;
    private Button btnCNumero;
    private Button btnBNumero;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtMostar = (EditText) findViewById(R.id.etMostar);
        btn1Numero = (Button) findViewById(R.id.btn1Numero);
        btn2Numero = (Button) findViewById(R.id.btn2Numero);
        btn3Numero = (Button) findViewById(R.id.btn3Numero);
        btn4Numero = (Button) findViewById(R.id.btn4Numero);
        btn5Numero = (Button) findViewById(R.id.btn5Numero);
        btn6Numero = (Button) findViewById(R.id.btn6Numero);
        btn7Numero = (Button) findViewById(R.id.btn7Numero);
        btn8Numero = (Button) findViewById(R.id.btn8Numero);
        btn9Numero = (Button) findViewById(R.id.btn9Numero);
        btnBNumero = (Button) findViewById(R.id.btnBNumero);
        btnCNumero = (Button) findViewById(R.id.btnCNumero);
        btn1Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn1Numero.getText());
            }
        });

        btn2Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn2Numero.getText());
            }
        });

        btn3Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn3Numero.getText());
            }
        });

        btn4Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn4Numero.getText());
            }
        });

        btn5Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn5Numero.getText());
            }
        });

        btn6Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn6Numero.getText());
            }
        });

        btn7Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn7Numero.getText());
            }
        });

        btn8Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn8Numero.getText());
            }
        });

        btn9Numero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Mostrar(btn9Numero.getText());
            }
        });

       btnCNumero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                MostrarC();
            }
        });

        btnBNumero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Borrar();
            }
        });
    }

    private void Borrar() {
        try {
            String cadena = edtMostar.getText().toString();

            cadena = cadena.substring(0, cadena.length() - 1);

            edtMostar.setText(cadena);
        }
        catch (Exception e){
            Toast.makeText(this,
                    "Introdueix un numero.",
                    Toast.LENGTH_SHORT).show();
        }
    }

    private void MostrarC() {
        try {
            edtMostar.setText("");
        }
        catch (Exception e){
            Toast.makeText(this,
                    "Introdueix un numero.",
                    Toast.LENGTH_SHORT).show();
        }
    }

    private void Mostrar(CharSequence text) {
        try {
            CharSequence Total = edtMostar.getText() + "" + text;
            edtMostar.setText(Total);
        }
        catch (Exception e){
            Toast.makeText(this,
                    "Introdueix un numero.",
                    Toast.LENGTH_SHORT).show();
        }
    }


}

File: CalculadoraKotlin/app/src/main/java/com/example/calculadorakotlin/MainActivity.kt
package com.example.calculadorakotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import java.util.stream.Collectors.toList

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private var action: actions? = null
    private var valorAnterior: String ? = null

    var tv : TextView? = null;
    var tvR : TextView? = null;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tv = findViewById<TextView>(R.id.tvOperacion)
        tvR = findViewById<TextView>(R.id.tvResultado)

        implement_buttons();
    }

    fun implement_buttons(){
        val button_0 = findViewById<Button>(R.id.btn_0);
        button_0.setOnClickListener { onClick(button_0) }

        val button_1 = findViewById<Button>(R.id.btn_1);
        button_1.setOnClickListener { onClick(button_1) }

        val button_2 = findViewById<Button>(R.id.btn_2);
        button_2.setOnClickListener { onClick(button_2) }

        val button_3 = findViewById<Button>(R.id.btn_3);
        button_3.setOnClickListener { onClick(button_3) }

        val button_4 = findViewById<Button>(R.id.btn_4);
        button_4.setOnClickListener { onClick(button_4) }

        val button_5 = findViewById<Button>(R.id.btn_5);
        button_5.setOnClickListener { onClick(button_5) }

        val button_6 = findViewById<Button>(R.id.btn_6);
        button_6.setOnClickListener { onClick(button_6) }

        val button_7 = findViewById<Button>(R.id.btn_7);
        button_7.setOnClickListener { onClick(button_7) }

        val button_8 = findViewById<Button>(R.id.btn_8);
        button_8.setOnClickListener { onClick(button_8) }

        val button_9 = findViewById<Button>(R.id.btn_9);
        button_9.setOnClickListener { onClick(button_9) }

        val btn_coma = findViewById<Button>(R.id.btn_coma);
        btn_coma.setOnClickListener { onClick(btn_coma) }

        val btn_mas = findViewById<Button>(R.id.btn_mas);
        btn_mas.setOnClickListener { onClick(btn_mas) }

        val btn_menos = findViewById<Button>(R.id.btn_menos);
        btn_menos.setOnClickListener { onClick(btn_menos) }

        val btn_mult = findViewById<Button>(R.id.btn_mult);
        btn_mult.setOnClickListener { onClick(btn_mult) }

        val btn_div = findViewById<Button>(R.id.btn_div);
        btn_div.setOnClickListener { onClick(btn_div) }

        val btn_result = findViewById<Button>(R.id.btn_result);
        btn_result.setOnClickListener { onClick(btn_result) }

        val btn_c = findViewById<Button>(R.id.btn_c);
        btn_c.setOnClickListener { onClick(btn_c) }

        val btn_del = findViewById<Button>(R.id.btn_del);
        btn_del.setOnClickListener { onClick(btn_del) }
    }

    fun Create_Operation(numOrcharacter: String?){
        if(!tv!!.text.isEmpty()) {
            if (numOrcharacter.equals(",") && !tv?.text.toString().contains(",")) {
                tv!!.setText(tv!!.text.toString() + numOrcharacter)
            }
            else if(!numOrcharacter.equals(","))
                tv!!.setText(tv!!.text.toString() + numOrcharacter)
        }
        else
            tv!!.setText(tv!!.text.toString() + numOrcharacter)
    }

    fun result() {
        var result: Float = 0f
        var string = !tv?.text.toString().isEmpty ? tv?.text.toString().replace(",",".") : "0"
        valorAnterior = valorAnterior?.replace(",",".")

        when (action){
            actions.SUMAR->{
                result = valorAnterior?.toFloat()!! + string.toFloat()
                tv?.setText( result.toString().replace(".", ","));
            }
            actions.RESTAR->{
                result = valorAnterior?.toFloat()!! - string.toFloat()
                tv?.setText( result.toString().replace(".", ","));
            }
            actions.MULTIPLICAR->{
                result = valorAnterior?.toFloat()!! * string.toFloat()
                tv?.setText( result.toString().replace(".", ","));
            }
            actions.DIVIDIR->{
                result = valorAnterior?.toFloat()!! / string.toFloat()
                tv?.setText( result.toString().replace(".", ","));
            }
        }
    }

    override fun onClick(p0: View?) {

        when(p0!!.id){
            R.id.btn_0 -> {
                Create_Operation("0")
            }
            R.id.btn_1 -> {
                Create_Operation("1")
            }
            R.id.btn_2 -> {
                Create_Operation("2")
            }
            R.id.btn_3 -> {
                Create_Operation("3")
            }
            R.id.btn_4 -> {
                Create_Operation("4")
            }
            R.id.btn_5 -> {
                Create_Operation("5")
            }
            R.id.btn_6 -> {
                Create_Operation("6")
            }
            R.id.btn_7 -> {
                Create_Operation("7")
            }
            R.id.btn_8 -> {
                Create_Operation("8")
            }
            R.id.btn_9 -> {
                Create_Operation("9")
            }
            R.id.btn_coma -> {
                Create_Operation(",")
            }
            R.id.btn_c -> {
                tv?.setText("")!!
                tvR?.setText("")!!
                action = null
            }
            R.id.btn_del -> {
                if(!tv?.text.toString()!!.isEmpty())
                    tv?.setText(tv?.text.toString().substring(0, tv?.text.toString().length - 1))
            }
            R.id.btn_coma -> {
                Create_Operation(",")
            }
            R.id.btn_mas -> {
                valorAnterior = tv?.text.toString()
                tv?.setText("")
                action = actions.SUMAR
                tvR?.setText("+")!!
            }
            R.id.btn_menos -> {
                valorAnterior = tv?.text.toString()
                tv?.setText("")
                action = actions.RESTAR
                tvR?.setText("-")!!
            }
            R.id.btn_mult -> {
                valorAnterior = tv?.text.toString()
                tv?.setText("")
                action = actions.MULTIPLICAR
                tvR?.setText("*")!!
            }
            R.id.btn_div -> {
                valorAnterior = tv?.text.toString()
                tv?.setText("")
                action = actions.DIVIDIR
                tvR?.setText("/")!!
            }
            R.id.btn_result -> {
               result()
                tvR?.setText("")
            }
        }
    }
}
